source/api/tests/test_api_integration.py

```diff
@@ -381,3 +382,83 @@ def test_create_upload_delete_dataset_DIMENSION_JSON_SUB_DIRECTORY(test_configs,
 
 def test_create_upload_delete_dataset_FACT_JSON_SUB_DIRECTORY(test_configs, test_data_set_type, get_etl_data_by_job_id, generate_random_test_files_to_s3_bucket):
    test_data_set_type("FACT", "JSON", test_configs, get_etl_data_by_job_id, generate_random_test_files_to_s3_bucket, s3_key_sub_dir="integ_test_data/")
+
+def test_create_upload_delete_dataset_DIMENSION_JSON_LIVE_RAMP(test_configs, test_data_set_type, get_etl_data_by_job_id, generate_random_test_files_to_s3_bucket):
+    test_columns = [
+        {
+            "name": "user_id",
+            "description": "The customer resolved id",
+            "dataType": "STRING",
+            "nullable": True,
+            "isMainUserId": True
+        },
+        {
+            "name": "user_type",
+            "description": "The customer resolved type",
+            "dataType": "STRING",
+            "nullable": True,
+            "isMainUserIdType": True
+        },
+        {
+            "name": "first_name",
+            "description": "hashed First name",
+            "dataType": "STRING",
+            "externalUserIdType": {
+            "type": "HashedIdentifier",
+            "identifierType": "FIRST_NAME"
+            },
+            "nullable": True
+        },
+        {
+            "name": "product_name",
+            "description": "Product name",
+            "dataType": "STRING",
+            "externalUserIdType": {
+                "type": "LiveRamp"
+            }
+        }
+    ]
+    test_data_set_type("DIMENSION", "JSON", test_configs, get_etl_data_by_job_id, generate_random_test_files_to_s3_bucket, test_columns=test_columns)
+
+def test_create_upload_delete_dataset_FACT_JSON_LIVE_RAMP(test_configs, test_data_set_type, get_etl_data_by_job_id, generate_random_test_files_to_s3_bucket):
+    test_columns = [
+        {
+            "name": "user_id",
+            "description": "The customer resolved id",
+            "dataType": "STRING",
+            "nullable": True,
+            "isMainUserId": True
+        },
+        {
+            "name": "user_type",
+            "description": "The customer resolved type",
+            "dataType": "STRING",
+            "nullable": True,
+            "isMainUserIdType": True
+        },
+        {
+            "name": "first_name",
+            "description": "hashed First name",
+            "dataType": "STRING",
+            "externalUserIdType": {
+            "type": "HashedIdentifier",
+            "identifierType": "FIRST_NAME"
+            },
+            "nullable": True
+        },
+        {
+            "name": "timestamp",
+            "description": "Timestamp",
+            "dataType": "TIMESTAMP",
+            "isMainEventTime": True
+        },
+        {
+            "name": "product_quantity",
```

Comment:
Same comment as above. A live ramp column will be a live ramp id, not a product quantity.

Reply:
Resolved